lib/atomic_time/attributes.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'active_support/duration'
+
+class AtomicTime
+  module Attributes
+    def initialize(secs)
+      raise TypeError unless secs.is_a? Numeric
+      raise ArgumentError if (secs = secs.to_i).negative?
```

Comment:
Why can't you do `if secs.to_i` instead of `(secs = secs.to_i)`, like why the assignment operator? Keeping in mind I'm not as familiar with POROs as I am with Rails.

Reply:
I wanted to ensure that `@val` is always stored as an integer internally to prevent any possible future headaches. The alternative would've been doing `secs = secs.to_i` above the guard clauses.

Reply:
Oh I see -- the fact that it's within the guard clause threw me off, you always want it to happen.